util/src/lib.rs

```diff
@@ -1 +1,34 @@
+use tokio::signal;
+use tokio_util::sync::CancellationToken;
 
+/// Returns a Future that triggers cancel_token and completes once a relevant signal to shutdown
+/// the service is caught.
+pub async fn signal_catcher(cancel_token: CancellationToken) {
+    // SIGTERM is the most important signal to handle since it is the one that is sent by
+    // systemd (as a result of commands like 'systemctl stop' or 'systemctl restart').
+    let sigterm = async {
+        signal::unix::signal(signal::unix::SignalKind::terminate())
+            .expect("failed to install signal handler for SIGTERM")
+            .recv()
+            .await;
+    };
+
+    // SIGINT could also be relevant in some cases
+    let ctrl_c = async {
+        signal::ctrl_c()
+            .await
+            .expect("failed to install signal handler for SIGINT (Ctrl-C)");
+    };
+
+    tokio::select! {
+        _ = sigterm => {},
+        _ = ctrl_c => {},
+    }
+
+    cancel_token.cancel()
+}
+
+/// Returns a Future that completes once cancel_token is triggered.
+pub async fn await_cancellation(cancel_token: CancellationToken) {
+    _ = cancel_token.cancelled().await;
+}
```

Comment:
This function seems unnecessary

Reply:
Agreed. There was some technical reason why I wrapped it. I'll take another look and see if I can solve the problem in a simpler way.

Reply:
Done!